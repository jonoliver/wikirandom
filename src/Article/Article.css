article {
  border-bottom: 1px solid #e2e2e2;
  padding-bottom: 10px;
  margin-bottom: 10px;
}

article p {
  text-align: left;
}
article img {
  max-width: 100%;
}
