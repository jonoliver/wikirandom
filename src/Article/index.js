import React from 'react';

const Article = ({article}) => {
  const { title, body, image } = article;
  return (
    <article>
      <h1>{title}</h1>
      <p>{body}</p>
      <img src={image} alt={image} />
    </article>
  )
}

export default Article;
