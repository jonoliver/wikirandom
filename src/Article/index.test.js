import React from 'react';
import ReactDOM from 'react-dom';
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Article from './index';

const div = document.createElement('div');

const article = {
  title: 'A title',
  body: 'A body',
  image: 'image.jpg'
}

it('renders without crashing', () => {
  ReactDOM.render(<Article article={article} />, div);
});

test('snapshots', () => {
  const component = Renderer.create(<Article article={article} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
