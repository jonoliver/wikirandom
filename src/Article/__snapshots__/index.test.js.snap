exports[`test snapshots 1`] = `
<article>
  <h1>
    A title
  </h1>
  <p>
    A body
  </p>
  <img
    alt="image.jpg"
    src="image.jpg" />
</article>
`;
